### DISCUSSION TITLE
ttkbootstrap error when I run Auto py to exe

### DISCUSSION THREAD

**Discussion by LesHol** [brentvollebregt/auto-py-to-exe#471] [Q&A] [answered]
Hi I am new to this so bear with me if this is a stupid question! I have all my files: main.py and four *.png files in the same directory. When I run this happens:

C:\>auto-py-to-exe
Warning: Installing shim for bottle import redirects (using Python=3.12.2 and bottle=0.12.25)
C:\Users\holml\AppData\Local\Programs\Python\Python312\Lib\site-packages\ttkbootstrap\localization\msgs.py:212: SyntaxWarning: invalid escape sequence '\$'
  'Cannot change to the directory %1\$s.\nPermission denied.',
C:\Users\holml\AppData\Local\Programs\Python\Python312\Lib\site-packages\ttkbootstrap\localization\msgs.py:213: SyntaxWarning: invalid escape sequence '\$'
  '无法更改目录 %1\$s。\n访问被拒绝。',
C:\Users\holml\AppData\Local\Programs\Python\Python312\Lib\site-packages\ttkbootstrap\localization\msgs.py:224: SyntaxWarning: invalid escape sequence '\$'
  ('Directory %1\$s does not exist.', '目录 %1\$s 不存在。'),
C:\Users\holml\AppData\Local\Programs\Python\Python312\Lib\site-packages\ttkbootstrap\localization\msgs.py:224: SyntaxWarning: invalid escape sequence '\$'
  ('Directory %1\$s does not exist.', '目录 %1\$s 不存在。'),
C:\Users\holml\AppData\Local\Programs\Python\Python312\Lib\site-packages\ttkbootstrap\localization\msgs.py:228: SyntaxWarning: invalid escape sequence '\$'
  ("Error: %1\$s", "错误： %1\$s"),
C:\Users\holml\AppData\Local\Programs\Python\Python312\Lib\site-packages\ttkbootstrap\localization\msgs.py:228: SyntaxWarning: invalid escape sequence '\$'
  ("Error: %1\$s", "错误： %1\$s"),
C:\Users\holml\AppData\Local\Programs\Python\Python312\Lib\site-packages\ttkbootstrap\localization\msgs.py:232: SyntaxWarning: invalid escape sequence '\$'
  'File %1\$s already exists.\nDo you want to overwrite it?',
C:\Users\holml\AppData\Local\Programs\Python\Python312\Lib\site-packages\ttkbootstrap\localization\msgs.py:233: SyntaxWarning: invalid escape sequence '\$'
  '文件 %1\$s 已经存在。\n您想要覆盖它吗？',
C:\Users\holml\AppData\Local\Programs\Python\Python312\Lib\site-packages\ttkbootstrap\localization\msgs.py:235: SyntaxWarning: invalid escape sequence '\$'
  ('File %1\$s already exists.\n\n', '文件 %1\$s 已经存在。\n\n'),
C:\Users\holml\AppData\Local\Programs\Python\Python312\Lib\site-packages\ttkbootstrap\localization\msgs.py:235: SyntaxWarning: invalid escape sequence '\$'
  ('File %1\$s already exists.\n\n', '文件 %1\$s 已经存在。\n\n'),
C:\Users\holml\AppData\Local\Programs\Python\Python312\Lib\site-packages\ttkbootstrap\localization\msgs.py:236: SyntaxWarning: invalid escape sequence '\$'
  ('File %1\$s does not exist.', '文件 %1\$s 不存在。'),
C:\Users\holml\AppData\Local\Programs\Python\Python312\Lib\site-packages\ttkbootstrap\localization\msgs.py:236: SyntaxWarning: invalid escape sequence '\$'
  ('File %1\$s does not exist.', '文件 %1\$s 不存在。'),
C:\Users\holml\AppData\Local\Programs\Python\Python312\Lib\site-packages\ttkbootstrap\localization\msgs.py:251: SyntaxWarning: invalid escape sequence '\$'
  ('Invalid file name %1\$s.', '无效的文件名 %1\$s。'),
C:\Users\holml\AppData\Local\Programs\Python\Python312\Lib\site-packages\ttkbootstrap\localization\msgs.py:251: SyntaxWarning: invalid escape sequence '\$'
  ('Invalid file name %1\$s.', '无效的文件名 %1\$s。'),
C:\Users\holml\AppData\Local\Programs\Python\Python312\Lib\site-packages\ttkbootstrap\validation.py:1: SyntaxWarning: invalid escape sequence '\d'
  """

Anyone with an idea how this can be fixed?

TIA Leslie

**Comment by brentvollebregt** [comment]
What are you doing to cause these messages?

If I had to guess, it looks like you are picking a file or directory and they are named with non-utf-8 characters?

**Comment by LesHol** [comment]
Hi
The directory where the files are named Password Manager 2.0. Could that be
it?

Leslie

Den fre 22 mars 2024 12:55Brent Vollebregt ***@***.***> skrev:

**Reply by brentvollebregt** [reply]
My first question again - what are you doing to cause these messages? Do they appear when you initially start the application or you do have to click on something?

I see these references point to [this file](https://github.com/israel-dryer/ttkbootstrap/blob/master/src/ttkbootstrap/localization/msgs.py#L213) - so it's not what's in your directory.

Also does this break the application? They look like just warnings, so could be ignored.

**Comment by LesHol** [comment]
Hi

The errors appear when I run auto-py-to-exe. I notice that on the second
line it says: 'Cannot change to the directory %1\$s.\nPermission denied.',
I have tried to run auto as admin but that didn't work either, I still get
the same errors.

I changed the name of the directory with the same result.

Regards


Leslie Holm
073-1802567
***@***.***

Södra Lyngbyvägen 52
29792 Everöd


Den fre 22 mars 2024 kl 21:41 skrev Brent Vollebregt <
***@***.***>:

**Comment by LesHol** [comment]
Hi

Ok maybe there is no problem with the warnings but I get an error when
trying to run the app. This is my setup. Everything is in the same
directory, including my *.png files.

[image: bild.png]
And here is the error:

[image: bild.png]
As you can see there are no *.png files. When I put them in this directory
it works. Is it meant to work this way or do I do something wrong? Then I
get another error, exit is not defined (which I saw some solution in your
"common problems").

Med vänlig hälsning


Leslie Holm
073-1802567
***@***.***

Södra Lyngbyvägen 52
29792 Everöd


Den fre 22 mars 2024 kl 21:41 skrev Brent Vollebregt <
***@***.***>:

**Reply by brentvollebregt** [reply]
Those images have not come through - you can paste the error here.

**Reply by LesHol** [reply]
![configuration](https://github.com/brentvollebregt/auto-py-to-exe/assets/28740804/4747da5c-137b-41a6-8edd-f33ea71c4a0b)
![error](https://github.com/brentvollebregt/auto-py-to-exe/assets/28740804/a5f76207-648e-4916-976f-d28179ab6718)

Hope this works!

**Reply by brentvollebregt** [reply]
That second screenshot has an error in it - the error takes up pretty much the whole dialog. In the help post that you hopefully read, there is a section ["FileNotFoundError: [Errno 2] No such file or directory"](https://nitratine.net/blog/post/issues-when-using-auto-py-to-exe/#filenotfounderror-errno-2-no-such-file-or-directory) - that gives some reasoning on why this has happened.

**Comment by LesHol** [comment]
I didn't get an errormessage at all! Are the files supposed to be copied to
the output directory or incorporated in the main file?
One of the times only one of the files was copied.



Den fre 29 mars 2024 08:20Brent Vollebregt ***@***.***> skrev:

**Reply by brentvollebregt** [reply]
In your previous comment, you said "And here is the error:" - what is in the image? The image you attached did not come through - [see here](https://github.com/brentvollebregt/auto-py-to-exe/discussions/471#discussioncomment-8918642).

**Comment by LesHol** [comment]
Ok I have enclosed the pictures now. Sorry for the inconvenience.


Med vänlig hälsning


Leslie Holm
073-1802567
***@***.***

Södra Lyngbyvägen 52
29792 Everöd


Den mån 1 apr. 2024 kl 11:32 skrev Brent Vollebregt <
***@***.***>:

**Reply by brentvollebregt** [reply]
They still have not come through from what I can see

**Comment by LesHol** [comment]
That is very strange because I enclosed them as separate files! Could it be
some security issue?

Den tis 2 apr. 2024 08:11Brent Vollebregt ***@***.***> skrev:

**Reply by brentvollebregt** [reply]
Are you trying to send them through by email? That may be the reason. You can add them to your comments on [https://github.com/brentvollebregt/auto-py-to-exe/discussions/471](https://github.com/brentvollebregt/auto-py-to-exe/discussions/471)

**Comment by LesHol** [comment]
Hi
I don't understand, why didn't the *.png files get copied?
Am I supposed to copy them manually?
In that case what are the "files"dialog supposed to do?



Den ons 3 apr. 2024 06:53Brent Vollebregt ***@***.***> skrev:

**Reply by brentvollebregt** [reply, marked as answer]
> I don't understand, why didn't the *.png files get copied?

This will find Python files - any extra asserts like images need to be added manually. See step 6 under "[General Steps](https://nitratine.net/blog/post/issues-when-using-auto-py-to-exe/#general-steps)" - "Add any extra files your project requires that aren't Python files (e.g. images, csv, databases, ...)"

> Am I supposed to copy them manually?

No, adding them in "Additional Files" will include them for you. I recommend packaging to "One Directory" first when adding additional files so you can see where they actually end up inside the bundle.

> In that case what are the "files"dialog supposed to do?

Are you referencing the "Additional Files" section? This adds non-Python files to the bundle output.

**Reply by LesHol** [reply]
Thank you Now I understand. Sorry about being so lazy. Now I have read all of your FAQ and it is very informative.
